src/fxapp/ReportManager.java

```diff
@@ -31,4 +31,14 @@ public void addReport(Report report) {
     public Stream<Report> getReports() {
         return reports.stream();
     }
+
+    public Report get(int i) {
+        if (reports == null) {
+            return null;
+        }
+        if (i >= reports.size()) {
+            return null;
+        }
```

Comment:
This whole method should be taken out IMO, but for future reference these if statements rang alarm bells in my head. You're doing extra checks that you don't need to -- this method is really just wrapping the `get` method of `ArrayList`, so we should let arraylist handle its own indexing (rather than explicitly doing it ourselves -- where we could make a mistake) and then if we need an invalid element to be null, we could just surround the reportManager.get(i) with a try catch were we're calling it.

```java
Report element;
try {
    element = reportManager.get(0);
} catch (IndexOutOfBoundsException e) { // or whatever the exception is called
    element = null;
}
// ...
```

Not trying to diss, just want to share I've picked up on after working on a lot of projects with people more talented than I am :nerd_face: